### PR TITLE
Ready For Testing - Implement callable functions

### PR DESCRIPTION
**Pull request by breningham**
Implements Firebase callable functions with a similar api as the web version 

Ref: 
https://firebase.google.com/docs/functions/callable

Closes: #806

### CODE REVIEW

**Comment by EddyVerbruggen** [comment]
Pretty stoked for this! I actually started playing with Cloud Functions last weekend and loved it, so I'd be happy to test and merge your efforts once you're happy with the implementation.

**Comment by breningham** [comment]
@EddyVerbruggen might be a good idea to have some basic HttpsCallableFunctions on the Firebase project for the Demo Applications

**Comment by EddyVerbruggen** [comment]
Good call(able)!

Would you like to have it in the non-ng demo app?

**Comment by EddyVerbruggen** [comment]
Can I have your Firebase username (to add you to the console)?

**Comment by breningham** [comment]
I'll Send it you on slack as i use my email.

**Comment by breningham** [comment]
Implementation is complete, ready for testing!

File: src/functions/functions.ios.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+    return (data: I ) => new Promise((resolve, reject) => {
+
+        const callable = functions.HTTPSCallableWithName(functionName);
+
+        const handleCompletion = (result: FIRHTTPSCallableResult, err: NSError) => {
+            if ( err ) {
+                if ( err.domain === FIRFunctionsErrorDomain ) {
+                    const message = err.localizedDescription;
+                    reject( message );
```
Comment (EddyVerbruggen) [inline src/functions/functions.ios.ts]: Note to self: add a return here, after merging.

File: src/functions/functions.ios.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+                }
+
+                reject( err.localizedDescription );
```
Comment (EddyVerbruggen) [inline src/functions/functions.ios.ts]: .. and here as well.

**Comment by EddyVerbruggen** [comment]
This is so perfect it hurts. Thanks a lot!

I'll be adding some documentation to wrap it up.

**Pull request merged**